cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -428,26 +430,30 @@ runTxBuild (AnyCardanoEra era) (AnyConsensusModeParams cModeParams) networkId mS
           <*> validateTxMintValue         era mValue
           <*> validateTxScriptValidity    era mScriptValidity
 
-      -- TODO: Combine queries
-      let localConnInfo = LocalNodeConnectInfo
-                            { localConsensusModeParams = CardanoModeParams (EpochSlots 21600)
-                            , localNodeNetworkId       = networkId
-                            , localNodeSocketPath      = sockPath
-                            }
-
       eInMode <- case toEraInMode era CardanoMode of
                    Just result -> return result
                    Nothing ->
                      left (ShelleyTxCmdEraConsensusModeMismatchTxBalance outBody
                             (AnyConsensusMode CardanoMode) (AnyCardanoEra era))
 
-      let utxoQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe
-                        (QueryUTxO . QueryUTxOByTxIn $ Set.fromList onlyInputs)
-      let pParamsQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryProtocolParameters
-      utxo <- executeQuery era cModeParams localConnInfo utxoQuery
-      pparams <- executeQuery era cModeParams localConnInfo pParamsQuery
-      (eraHistory, systemStart) <- firstExceptT ShelleyTxCmdAcquireFailure
-                                     $ newExceptT $ queryEraHistoryAndSystemStart localNodeConnInfo Nothing
+      (utxo, pparams, eraHistory, systemStart, stakePools) <-
```

Comment:
@newhoggy  I originally suggested changing from `ExceptT _ IO _` to `IO (Either _ _)` in the return type of `executeLocalStateQueryExpr`, was that meant to avoid some of this pain here?

I'll leave it up to you, but you could also just replace `joinEitherM ShelleyTxCmdAcquireFailure id` with `fmap (join . first ShelleyTxCmdAcquireFailure)` then you can avoid having to create `joinEither`/`joinEitherM` helper functions.

Reply:
Lets move those helper functions to the cli or refactor as David suggested above.

Reply:
Refactored as suggested.